src/v/storage/kvstore.h

```diff
@@ -99,6 +99,7 @@ class kvstore {
         consensus = 1,
         storage = 2,
         controller = 3,
+        coproc = 4,
```

Comment:
i don't follow this change. a write to the kvstore is blocking.... the coproc does not need to block. 

the kvstore doesn't need to wait at all, it doesn't even need to return a future.